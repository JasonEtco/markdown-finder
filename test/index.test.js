const finder = require('../lib')

describe('MarkdownFinder', () => {
  describe('has', () => {
    describe('image', () => {
      it('returns true if the markdown contains an image', () => {
        const md = 'This is an image ![img](example.com/image.png)'
        expect(finder(md).has.image()).toBe(true)
      })

      it('returns false if the markdown does not contain an image', () => {
        const md = 'This is not an image'
        expect(finder(md).has.image()).toBe(false)
      })

      it('does not match non-image links', () => {
        const md = 'This is a link [link](url)'
        expect(finder(md).has.image()).toBe(false)
      })
    })

    describe('link', () => {
      it('returns true if the markdown contains an link', () => {
        const md = 'This is a link [link](example.com)'
        expect(finder(md).has.link()).toBe(true)
      })

      it('returns false if the markdown does not contain a link', () => {
        const md = 'This is not an link'
        expect(finder(md).has.link()).toBe(false)
      })

      it('does not match image links', () => {
        const md = 'This is an image ![img](url)'
        expect(finder(md).has.link()).toBe(false)
      })
    })
  })

  describe('get', () => {
    describe('image', () => {
      it('returns the image if it exists', () => {
        const md = 'This is an image ![img](example.com/image.png)'
        expect(finder(md).get.image()).toMatchSnapshot()
      })

      it('returns false if it does not exist', () => {
        const md = 'This is not an image'
        expect(finder(md).get.image()).toBe(false)
      })
    })

    describe('link', () => {
      it('returns true if the markdown contains an link', () => {
        const md = 'This is a link [link](example.com)'
        expect(finder(md).get.link()).toMatchSnapshot()
      })

      it('returns false if the markdown does not contain a link', () => {
        const md = 'This is not an link'
        expect(finder(md).get.link()).toBe(false)
      })
    })
  })
})
