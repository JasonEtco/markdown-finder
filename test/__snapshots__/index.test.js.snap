// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownFinder images returns an array of images 1`] = `
Array [
  Object {
    "match": "![img](example.com/image.png)",
    "text": "img",
    "url": "example.com/image.png",
  },
  Object {
    "match": "![img](example.com/photo.png)",
    "text": "img",
    "url": "example.com/photo.png",
  },
]
`;

exports[`MarkdownFinder links returns an array of links 1`] = `
Array [
  Object {
    "match": "[link](example.com)",
    "text": "link",
    "url": "example.com",
  },
  Object {
    "match": "[this](this.com)",
    "text": "this",
    "url": "this.com",
  },
]
`;
